### PR TITLE
MOSIP-27456

### PR DESCRIPTION
**Pull request by JanardhanBS-SyncByte**
CTK Test case Abis-3028
to check if referenceId not found (in ABIS) in Gallery throw error 12

### CODE REVIEW

File: mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java [hunk 1]
```diff
@@ -68,8 +68,8 @@ public class ProxyAbisInsertServiceImpl implements ProxyAbisInsertService {
 
 	private static final Logger logger = LoggerFactory.getLogger(ProxyAbisInsertServiceImpl.class);
 
-	private static String UPLOAD_FOLDER = System.getProperty("user.dir")+"\\keystore";
-	private static String PROPERTIES_FILE = UPLOAD_FOLDER+ "\\partner.properties";
+	private static String UPLOAD_FOLDER = System.getProperty("user.dir") + "\\keystore";
```
Comment (gsasikumar) [inline mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java:71]: We should never append the file with a +

Use this approach 
'''Path currentPath = Paths.get(System.getProperty("user.dir"));
Path filePath = Paths.get(currentPath.toString(), "data", "foo.txt");
'''

File: mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java [hunk 2]
```diff
@@ -390,24 +365,35 @@ public IdentifyDelayResponse findDuplication(IdentityRequest ir) {
...
 					&& !ir.getGallery().getReferenceIds().get(0).getReferenceId().isEmpty()) {
 				List<String> referenceIds = new ArrayList();
 				ir.getGallery().getReferenceIds().stream().forEach(ref -> referenceIds.add(ref.getReferenceId()));
 
 				logger.info("checking for duplication of reference Id against" + referenceIds.toString());
+
+				int galleryRefIdCountInDB = proxyAbisBioDataRepository
+						.fetchCountForReferenceIdPresentInGallery(referenceIds);
+				if (galleryRefIdCountInDB != referenceIds.size()) {
+					logger.info("checking for reference Id Present in DB ==== " + galleryRefIdCountInDB);
```
Comment (gsasikumar) [inline mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java:378]: Don't use logger with +

File: mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java [hunk 3]
```diff
@@ -536,13 +523,12 @@ private IdentityResponse.Analytics getAnalytics() {
...
-
-	public  String saveUploadedFileWithParameters(MultipartFile uploadedFile, String alias,
-												  String password, String keystore) {
+	public String saveUploadedFileWithParameters(MultipartFile uploadedFile, String alias, String password,
+			String keystore) {
 		try {
 			logger.info("Uploading certificate");
 			byte[] bytes = uploadedFile.getBytes();
-			Path path = Paths.get(UPLOAD_FOLDER + "/"+ uploadedFile.getOriginalFilename());
+			Path path = Paths.get(UPLOAD_FOLDER + "/" + uploadedFile.getOriginalFilename());
```
Comment (gsasikumar) [inline mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java:531]: Do not use the the +

**Review by gsasikumar — CHANGES_REQUESTED** [review]

File: mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java (hunk 1 above)
Comment (JanardhanBS-SyncByte) [reply mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java]: private static Path currentPath = Paths.get(System.getProperty("user.dir"));
private static Path keystoreFilePath = Paths.get(currentPath.toString(), "keystore");

File: mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java (hunk 2 above)
Comment (JanardhanBS-SyncByte) [reply mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java]: logger.info(String.format("checking for reference Id Present in DB[%d], Gallery reference Id list size[%d] ", galleryRefIdCountInDB, referenceIds.size()));

File: mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java (hunk 3 above)
Comment (JanardhanBS-SyncByte) [reply mock-abis/src/main/java/io/mosip/proxy/abis/service/impl/ProxyAbisInsertServiceImpl.java]: Path path = Paths.get(keystoreFilePath.toString(), uploadedFile.getOriginalFilename());

**Review by gsasikumar — APPROVED** [review]

**Pull request merged**